packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +26,17 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, path, options = {}) => {
+const matchPath = (pathname, path, options = {}, parentMatch = null) => {
   const { exact = false, strict = false } = options
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  const absolute = isAbsolute(path)
+
+  if (path === undefined)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
```

Comment:
Shouldn't this be `parentMatch ? parentMatch.url : pathname`? The intent here is that `<Route>`s with no `path` should report they matched the entire `pathname`. But maybe you have a reason for using `/`?